src/operator/operator_common.h

```diff
@@ -253,6 +300,48 @@ inline bool type_assign(int *y, const int& x) {
   }
 #endif
 
+/*! \brief assign stype to target_stype, if successful,
+ *         assign dispatch_mode to target_dispatch
+ */
+inline bool storage_type_assign(int* stype,
+                                const NDArrayStorageType target_stype,
+                                DispatchMode* dispatch,
+                                const DispatchMode target_dispatch) {
```

Comment:
why not disentangle this into two functions, storage_type_assign and dispatch_mode_assign? Looks like the logic is well separated

Reply:
Usually you'll check whether the stype_assign is successful or not before assigning dispatch mode. If you want to separate the two, it makes all infer_storage function very lengthy. Such logic is common and I think we should use one function for convenience. I still think the original name `dispatch_on_storage` is better